server/src/main/java/org/elasticsearch/cluster/metadata/ComponentTemplate.java

```diff
@@ -177,7 +177,7 @@ public XContentBuilder toXContent(
     ) throws IOException {
         builder.startObject();
         builder.field(TEMPLATE.getPreferredName());
-        this.template.toXContent(builder, params, rolloverConfiguration, globalRetention);
+        this.template.toXContent(builder, params, rolloverConfiguration, globalRetention, false);
```

Comment:
Are component templates always non-system?

Reply:
I believe that they are neither system or non-system -- they're just a component. I picked `false` since there is no system index related, and so that it would always show the global/factory effective retention when you're just looking at a component, but then if you put it into a composite template for a system data stream it would hide that. I think that's correct, but I'm not 100% sure.

Reply:
Actually I was just talking to @dakrone and he made the point that we ought to never show effective date at all in any template anyway, since it's not something you can update on a template. And I believe that we do not currently show effective date in templates because we don't put the `INCLUDE_EFFECTIVE_RETENTION_PARAM_NAME` param in parms here anyway. Kind of an odd flow of control. So the `isSystemDataStream` arg is irrelevant here. I'll try to make that more clear somehow -- maybe a version that doesn't even have that argument?